batch/inference_job.py

```diff
@@ -79,6 +82,44 @@ def launch_batch(config_file, num_jobs, sims_per_job, num_blocks, outputs, s3_bu
     return rc
 
 
+def autodetect_params(config, *, num_jobs=None, sims_per_job=None, num_blocks=None):
+    if num_jobs and sims_per_job and num_blocks:
+        return (num_jobs, sims_per_job, num_blocks)
+
+    if "filtering" not in config or "simulations_per_slot" not in config["filtering"]:
+        raise click.UsageError("filtering::simulations_per_slot undefined in config, can't autodetect parameters")
+    sims_per_slot = int(config["filtering"]["simulations_per_slot"])
+
+    if num_jobs is None:
+        num_jobs = config["nsimulations"]
+        print(f"Setting number of output slots to {num_jobs} [via config file]")
+
+    if sims_per_job is None:
+        if num_blocks is not None:
+            sims_per_job = int(math.ceil(sims_per_slot / num_blocks))
+            print(f"Setting number of blocks to {num_blocks} [via num_blocks (-k) argument]")
+            print(f"Setting sims per job to {sims_per_job} [via {sims_per_slot} simulations_per_slot in config]")
+        else:
+            geoid_fname = pathlib.Path(config["spatial_setup"]["base_path"]) / config["spatial_setup"]["geodata"]
+            with open(geoid_fname) as geoid_fp:
+                num_geoids = sum(1 for line in geoid_fp)
+
+            # formula based on a simple regression of geoids (based on known good performant params)
+            sims_per_job = max(50 - num_geoids // 72, 10)
+            sims_per_job = 5 * int(math.ceil(sims_per_job / 5))  # multiple of 5
+
+            num_blocks = int(math.ceil(sims_per_slot / sims_per_job))
+
+            print(f"Setting sims per job to {sims_per_job} [estimated based on {num_geoids} geoids]")
+            print(f"Setting number of blocks to {num_blocks} [via math]")
```

Comment:
Might be nice to print a recalculated sims_per_slot here since it's not necessarily equal to the one in the config.

Reply:
the sims_per_slot are rewritten [later in the pipeline as part of the run](https://github.com/HopkinsIDD/COVIDScenarioPipeline/blob/4baea48e663d0a1855536f20a593f35b0df94321/batch/inference_job.py#L54), it's used as an unchanged input var here

Reply:
that works for me then.  i guess if someone really wants to know that multiplication isn't too hard anyway

Reply:
added extra debug info as per feedback:
```
Setting number of output slots to 1000 [via config file]
Setting sims per job to 50 [estimated based on 166 geoids and 250 sims per slot in config]
Setting number of blocks to 5 [via math]
```

Reply:
I think instead of `{sims_per_job}` it should be `{num_blocks * sims_per_block}` unless i'm misreading?

Reply:
as an example, if i said 1000 sims_per_job initially, 15 sims per block, i'd expect to see 1005 sims_per job printed here